src/OutputWriters/jld2_writer.jl

```diff
@@ -164,6 +164,8 @@ function JLD2Writer(model, outputs; filename, schedule,
     outputs = NamedTuple(Symbol(name) => construct_output(outputs[name], model.grid, indices, with_halos)
                          for name in keys(outputs))
 
+    schedule = deepcopy(schedule) # assume user wants to reuse a schedule
+
     # Convert each output to WindowedTimeAverage if schedule::AveragedTimeWindow is specified
     schedule, outputs = time_average_outputs(schedule, outputs, model)
```

Comment:
Should we do it also for this line, if the user uses `AveragedTimeWindow`?

Reply:
what you suggest is just a different way of writing the same code, isn't it?